src/util/CartesianUtils.ts

```diff
@@ -32,15 +33,47 @@ export const formatAxisMap = (props: any, axisMap: any, offset: any, axisType: A
     const { orientation, domain, padding = {}, mirror, reversed } = axis;
     const offsetKey = `${orientation}${mirror ? 'Mirror' : ''}`;
 
-    let range, x, y, needSpace;
+    let calculatedPadding, range, x, y, needSpace;
+
+    if (axis.type === 'number' && (axis.padding === 'gap' || axis.padding === 'no-gap')) {
+      const diff = domain[1] - domain[0];
+      let smallestDistanceBetweenValues = Infinity;
+      const sortedValues = axis.categoricalDomain.sort();
+      sortedValues.forEach((value: number, index: number) => {
+        if (index > 0) {
+          smallestDistanceBetweenValues = Math.min(
+            (value || 0) - (sortedValues[index - 1] || 0),
+            smallestDistanceBetweenValues,
+          );
+        }
+      });
+      const smallestDistanceInPercent = smallestDistanceBetweenValues / diff;
+      const rangeWidth = axis.layout === 'vertical' ? offset.height : offset.width;
+
+      if (axis.padding === 'gap') {
```

Comment:
`smallestDistanceBetweenValues` should not be `Infinity`

Reply:
This is the same behaviour as for ticks, how would you avoid this?